src/main/scala/com/fulcrumgenomics/util/DemuxFastqs.scala

```diff
@@ -0,0 +1,579 @@
+/**
+  * Copyright (c) 2016, Fulcrum Genomics LLC
+  * All rights reserved.
+  *
+  * Redistribution and use in source and binary forms, with or without
+  * modification, are permitted provided that the following conditions are met:
+  *
+  * 1. Redistributions of source code must retain the above copyright notice,
+  * this list of conditions and the following disclaimer.
+  *
+  * 2. Redistributions in binary form must reproduce the above copyright notice,
+  * this list of conditions and the following disclaimer in the documentation
+  * and/or other materials provided with the distribution.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+  * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  * POSSIBILITY OF SUCH DAMAGE.
+  */
+package com.fulcrumgenomics.util
+
+import java.io.File
+import java.nio.file.Path
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, JeanLucTool}
+import com.fulcrumgenomics.util.miseq.{Sample, SampleBarcode, SampleSheet}
+import dagr.commons.CommonsDef.{DirPath, FilePath, PathToFastq, _}
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SAMFileHeader.SortOrder
+import htsjdk.samtools._
+import htsjdk.samtools.fastq.{FastqReader, FastqRecord}
+import htsjdk.samtools.metrics.MetricsFile
+import htsjdk.samtools.util._
+import picard.illumina.ExtractIlluminaBarcodes
+import picard.illumina.parser.ReadStructure
+import picard.util.IlluminaUtil
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+object DemuxFastqs {
+  val UnmatchedSampleId: String = "unmatched"
+  private val solexaQualityConverter: SolexaQualityConverter = SolexaQualityConverter.getSingleton
+
+  // TODO: duplicated but private in picard.sam.FastqToSam so make it public and static there, and use it here
+  /** Creates the read name to put into a SAM record from a FASTQ record. */
+  private def getReadName(fastqHeader: String, paired: Boolean, stripUnpairedMateNumber: Boolean): String = {
```

Comment:
I've always found this method super confusing in Picard - so if it is paired we shouldn't remove the mate number?  I think we should always remove the mate number, no _if_s, _and_s or _but_s.

Perhaps this instead:
```scala
def extractName(header: String): String = header.takeWhile(ch => ch != ' ').stripSuffix("/1").stripSuffix("/2")
```